### DISCUSSION TITLE
Widget samples not compiling in v4

### DISCUSSION THREAD

**Discussion by stevegdavis** [Mapsui/Mapsui#2549] [Q&A]
Hi, I am trying to add a Button Widget to my app. I have copied and pasted some code from the latest Mapsui samples but I cannot build as I get errors such as:-

WidgetEventArgs missing
also ButtonWidget does not contain a definition for Margin,Padding,Tapped

I have also seen that the samples have the following 

> using Mapsui.Widgets.ButtonWidgets; 

but I cannot use that, only 

> using Mapsui.Widgets.ButtonWidget; 

So I am confused which is correct.

I am using the latest nugets:-
Mapsui.Maui Version=4.1.4
Mapsui.Rendering.Skia Version=4.1.4
Mapsui.Tiling Version=4.1.4
and .NET8

Thanks for any help.

**Comment by pauldendulk** [comment]
The samples on the website correspond to main which contains the upcoming v5 code. You have to check out the develop/4.1 branch, and find the samples in there. You could run Mapsui.Samples.Blazor to see the equivalent of the online samples.

This a problem that will hit most users. I need to fix it. Step one is to publish only the v4 samples and docs to the website. Next I need to figure out of to deploy v5 alongside v4.

**Comment by stevegdavis** [comment]
OK, thanks for your reply.

**Comment by pauldendulk** [comment]
I resolved this today. The default website is now v4:
https://mapsui.com/
Samples:
https://mapsui.com/samples/

The v5 version of everything is in a v5 folder:
https://mapsui.com/v5/
Samples:
https://mapsui.com/v5/samples/

@inforithmics It was some trouble getting this to work. I fixed this with a lot of commits directly to main and develop/4.1. The solution is to use another deploy tool (which used the gh-pages branch, instead of artifacts) and which does not delete existing files. This way the the deployment of 4.1 does not replace the one of main or the other way around.

This was an issue that had been in the back of my mind for a while because I had no plan to fix it. Happy this is now done. I used peaceiris/actions-gh-pages@v3. Thanks @peaceiris

**Reply by pauldendulk** [reply]
I had to upgrade the 4.1 Blazor samples to .NET 8 because because of some sort of corruption that apparently resulted from the deployment process: 

```
failed to find a valid digest in the 'integrity' attribute for resource 'https://mapsui.com/samples/_framework/dotnet.7.0.17.fgb96quo4q.js' with computed SHA-256 integrity 'XguAFn2P9FXRogx5Kc4rFBXgPoayGUOmkazh5BVPivU='. The resource has been blocked.
```

**Reply by pauldendulk** [reply]
I created an issue for this two year ago. https://github.com/Mapsui/Mapsui/issues/1561

**Comment by stevegdavis** [comment]
Not sure if the V5 samples are a work in progress, but not all the blazor samples work particually the Info section, the map just freezes or is blank.

**Reply by pauldendulk** [reply]
Thanks for mentioning that. It seems to crash specifically on CustomCalloutSample, and after it crashes nothing else works anymore.

That sample works correctly on localhost, so I think the problem is with our deployment. Perhaps something got stripped away or altered (I have seen this with line ending being changed).

**Reply by pauldendulk** [reply]
It is fixed now. The json parsing failed.
